src/http/Headers.php

```diff
@@ -184,12 +185,18 @@ public function send($code = null, array $headers = null)
         }
 
         $body = $this->body;
-
-        if (array_key_exists('Content-Encoding', $this->headers)) {
-            $body = self::encodeBody($body, $this->headers['Content-Encoding']);
-            $this->headers['Content-Length'] = strlen($body);
+        $this->headers['Content-Length'] = strlen($body);
+
+        if ($compress === true) {
+            $encoding = '';
+            if (preg_match('/ *gzip *,?/', $this->headers['accept-encoding'])) {
+                $encoding = 'gzip';
+            } elseif (preg_match('/ *deflate *,?/', $this->headers['accept-encoding'])) {
+                $encoding = 'deflate';
+            }
+            (!empty($encoding)) ? $this->headers['Content-Encoding'] = $encoding : null;
```

Comment:
Do you consider using the `isset`?

```php
(isset($encoding)) ? $this->headers['Content-Encoding'] = $encoding : null;
```

This will be more readable than `!empty($encoding)`.

Reply:
thanks, yes sure